tests/test_app.py

```diff
@@ -200,35 +207,71 @@ async def test_cmd(txt):
 
 
 async def test_auth():
-    async with await AppClient.create(config) as cli:
+    async with await AppClient.create(config, TEST_TOKEN) as cli:
+        resp = await cli.get("/test-auth")
+        assert resp.status == 200
+        text = await resp.text()
+        assert f"I'm authenticated as {TEST_USER}" in text
+
+@pytest.mark.parametrize("cookie_name", ["kbase_session", "kbase_session_backup"])
+async def test_auth_cookies(cookie_name):
+    async with await AppClient.create(config, None, cookies={cookie_name: TEST_TOKEN}) as cli:
         resp = await cli.get("/test-auth")
         assert resp.status == 200
         text = await resp.text()
-        assert "I'm authenticated as" in text
+        assert f"I'm authenticated as {TEST_USER}" in text
+
+
+@pytest.mark.parametrize("token", [None, ""])
+async def test_auth_fail_no_token(token):
+    async with await AppClient.create(config, token) as cli:
+        resp = await cli.get("/test-auth")
+        assert resp.status == 401
+        text = await resp.text()
+        assert "must provide an auth token" in text
+
+
+async def test_auth_fail_bad_token():
+    async with await AppClient.create(config, "bad token") as cli:
+        resp = await cli.get("/test-auth")
+        assert resp.status == 401
+        text = await resp.text()
+        assert "token is invalid" in text
+
+
+@pytest.mark.parametrize("cookie_name", ["kbase_session", "kbase_session_backup"])
+async def test_auth_fail_bad_cookie_token(cookie_name):
+    async with await AppClient.create(config, None, cookies={cookie_name: "bad_token"}) as cli:
+        resp = await cli.get("/test-auth")
+        assert resp.status == 401
+        text = await resp.text()
+        assert "token is invalid" in text
+
+
+async def test_auth_fail_service_err():
+    # Not sure how to test this, as the service should fail on startup with a bad auth URL
+    # TODO: configure a local auth server with bad data to test this
+    pass
 
 
 async def test_service():
-    async with await AppClient.create(config) as cli:
+    async with await AppClient.create(config, TEST_TOKEN) as cli:
         resp = await cli.get("/test-service")
         assert resp.status == 200
         text = await resp.text()
         assert "staging service version" in text
 
 
-async def test_jbi_metadata():
+async def test_jgi_metadata():
     txt = "testing text\n"
-    username = "testuser"
```

Comment:
This pattern repeats for most of the rest of the file:
* remove `username = "testuser"` lines
* "testuser" now becomes TEST_USER extracted from the env var
* the auth token is now the second parameter passed to `AppClient.create`, so that's updated throughout.

The rest is replacing either the `username` variable or the hard-coded `"testuser"` string with `TEST_USER` everywhere.

Reply:
Also deleting `headers={"Authorization": ""}` from all the calls.